### ISSUE TITLE
React 15.2.1 throws unknown prop warning when using ...props

### ISSUE THREAD

**Issue by andrewhl** [azmenak/react-stripe-checkout#32]
```javascript
warning.js:44Warning: Unknown props `token`, `email`, `currency`, `description`, `panelLabel`, `image`, `allowRememberMe`, `stripeKey`, `locale`, `componentClass` on <span> tag. Remove these props from the element. For details, see https://fb.me/react-unknown-prop
    in span (created by ReactStripeCheckout)
```

The error is documented [here](https://fb.me/react-unknown-prop).

The issue is that you are passing `...this.props` into the ComponentClass. In my case, the element is a span, but these are non-standard attributes and unrecognized by that element. 

```javascript
  render: function () {
    var ComponentClass = this.props.componentClass;
    return (
      !this.props.children ? this.renderStripeButton() : (
        <ComponentClass {...this.props} onClick={this.onClick}>
          {this.props.children}
        </ComponentClass>
      )
    );
  }
```

I'm not sure what the correct fix for this is. Perhaps you need to explicitly define the above unrecognized props as data attributes (per [this documentation](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Using_data_attributes)). The react documentation on the warning provides a few possible solutions.

**Comment by stueynet** [comment]
+1

**Comment by traghav** [comment]
+1

**Comment by ratiotile** [comment]
+1

**Comment by Themandunord** [comment]
+1

**Comment by mikey0000** [comment]
+1

**Comment by azmenak** [comment]
This has been fixed in version 2.0 :)
